packages/ethereum-contracts/scripts/deploy-framework.js

```diff
@@ -451,17 +451,24 @@ module.exports = eval(`(${S.toString()})({skipArgv: true})`)(async function (
         }
     }
 
-    // deploy CFAv1Forwarder
-    await deployAndRegisterContractIf(
-        CFAv1Forwarder,
-        "CFAv1Forwarder",
-        async (contractAddress) => contractAddress === ZERO_ADDRESS,
-        async () => {
-            const forwarder = await CFAv1Forwarder.new(superfluid.address);
-            output += `CFA_V1_FORWARDER=${forwarder.address}\n`;
-            return forwarder;
-        }
-    );
+    // deploy CFAv1Forwarder for test deployments
```

Comment:
hmm, I am a bit confused by namings then, and I am a bit of out dated now on what assorted deployment scripts we have and what are they for.

what's your thinking behind it currently? @d10r

Reply:
* deploy-framework.js
* deploy-test-environment.js 
* deploy-test-framework.js

Reply:
still the same as before actually.. but maybe it's time for us to actually clarify it more (also taking into account backward compatibility if needed, or abandon it if unhelpful) since I have forgotten many things since then.

Reply:
Looks like `deploy-framework` is called in `deploy-test-environment`. 

From doing a quick search, we use `deploy-test-environment.js` for deployments to testnets and `deploy-framework.js` for mainnet as well as our contracts test suite (TestEnvironment.ts).

The `deploy-test-framework.ts` is used for deploying SF in SDK-Core/Subgraph using the `SuperfluidFrameworkDeployer`: https://github.com/superfluid-finance/protocol-monorepo/pull/1104

Reply:
This has left me a bit head-scratching too - the forwarder is kind of a special case, bcs not really part of the protocol, but plugged in via forwarder mechanism.

My understanding is:
* `deploy-framework.js` wants to be the protocol deploy script for all: mainnets (initial & upgrade), testnets (initial & upgrade), devnets.  
* `deploy-test-environment.js` is for testnets (initial & upgrade= and devnets, it additionally (to the protocol) deploys (or upgrades to latest logic) test tokens.
* `deploy-test-framework.ts` uses - as @0xdavinchee already mentioned - `SuperfluidFrameworkDeployer`, which isn't as self-sufficient as I believed it is. It seems to have no truffle dependency (?), thus may be a useful basis for hardhat projects. I just had that topic with Sam, discussing what the canonical way of deploying for tests in our example projects should be.

Reply:
Will mark this as resolved for now and create a new issue for classifying/cleaning up these scripts.